ParallaxGenLib/src/NIFUtil.cpp

```diff
@@ -204,20 +204,26 @@ auto NIFUtil::setShaderType(nifly::NiShader *NIFShader, const nifly::BSLightingS
   if (NIFShader->GetShaderType() != Type) {
     NIFShader->SetShaderType(Type);
     Changed = true;
+  } else {
+    Changed = false;
```

Comment:
Every other function that calls these functions expects changed to **not** be set to false explicitly if the change did not occur. This is because the caller sets Changed to false and then calls however many nifutil methods are needed so changed gets set to true by the individual methods if needed, but the default is false. In other words changed is meant to be a global changed for the whole shape in this context.

In general I don't have a problem with changing that logic to what you have here, in fact it's probably more readable/understandable this way, but the logic for the upstream callers needs to be updated in that case. Mainly the patchers call these methods I believe.

Reply:
Oh, I never would have thought that this is intentional, I will have a look